File: src/stores/dumps.ts
import { create } from "zustand";

export interface DumpEntry {
  id: string;
  type: "dump" | "query" | "job" | "view" | "http" | "log";
  content: string;
  file: string | null;
  line: number | null;
  site: string | null;
  timestamp: string;
}

interface DumpsState {
  dumps: DumpEntry[];
  setDumps: (dumps: DumpEntry[]) => void;
  addDump: (dump: DumpEntry) => void;
  clearDumps: () => void;
}

export const useDumpsStore = create<DumpsState>((set) => ({
  dumps: [],
  setDumps: (dumps) => set({ dumps }),
  addDump: (dump) => set((state) => ({ dumps: [dump, ...state.dumps] })),
  clearDumps: () => set({ dumps: [] }),
}));

File: src-tauri/src/commands/dumps.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct DumpEntry {
    pub id: String,
    pub timestamp: String,
    pub dump_type: DumpType,
    pub content: String,
    pub file: Option<String>,
    pub line: Option<u32>,
    pub site: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum DumpType {
    Dump,
    Query,
    Job,
    View,
    HttpRequest,
    Log,
}

#[tauri::command]
pub async fn get_dumps() -> Result<Vec<DumpEntry>, String> {
    // TODO: Return dumps from the ring buffer
    Ok(Vec::new())
}

#[tauri::command]
pub async fn clear_dumps() -> Result<(), String> {
    // TODO: Clear the dump ring buffer
    Ok(())
}
